### PR TITLE
Notification wiring

### PR DESCRIPTION
**Pull request by zyro**

### CODE REVIEW

File: server/core_friend.go [hunk 1]
```diff
@@ -18,29 +18,73 @@ import (
...
+			}
+			var recipient []byte
+			var sender []byte
+			var subject string
+			var code int64
+			if isFriendAccept {
+				recipient = userID
+				sender = friendID
+				subject = fmt.Sprintf("%v accepted your friend request", handle)
+				code = 3
```
Comment (mofirouz) [inline server/core_friend.go]: Can we use some sort of an Enum here to indicate the message code, so it's not a magic number?

File: server/pipeline_friend.go [hunk 2]
```diff
@@ -100,7 +103,39 @@ func (p *pipeline) addFacebookFriends(logger *zap.Logger, userID []byte, accessT
 				if err != nil {
 					logger.Error("Could not commit transaction", zap.Error(err))
 				} else {
-					logger.Info("Imported friends from Facebook")
+					logger.Debug("Imported friends from Facebook")
+
+					// Send out notifications.
+					if len(friendUserIDs) != 0 {
+						content, err := json.Marshal(map[string]interface{}{"handle": handle, "facebook_id": fbid})
```
Comment (mofirouz) [inline server/pipeline_friend.go]: Should this also include the user_id as well? I know they can query the accounts by handle, I'm just playing devil's advocate here

Comment (zyro) [inline server/pipeline_friend.go]: The sender always contains the User ID anyway, so no need to add it to the content. The only things added to the content are data fields that don't already appear in the top-level notification fields or are embedded in the subject: "foo wants to be your friend" -> content is `{"handle": "foo"}`. Enough information to display a custom/localised notification as needed.

File: server/core_friend.go (hunk 1 above)
Comment (zyro) [inline server/core_friend.go]: We might be able to use one internally but the benefit will be minimal as we still need to allow arbitrary custom codes from developers.

Comment (mofirouz) [inline server/core_friend.go]: I think this is importantly for maintainability. Let's add this Enum in.

File: server/pipeline_friend.go (hunk 2 above)
Comment (mofirouz) [inline server/pipeline_friend.go]: Fair enough

**Pull request closed without merging**